soh/src/code/z_parameter.c

```diff
@@ -2782,6 +2782,38 @@ s32 Inventory_HasSpecificBottle(u8 bottleItem) {
     }
 }
 
+#define RBA_EQUIPMENT_OFFSET 0x28
+#define RBA_UPGRADE_OFFSET 0x2C
+#define RBA_QUEST_OFFSET 0x30
+#define RBA_GS_OFFSET 0x5C
+
+void byteSwapInventory() {
+    u8 equipment[2];
+    u8 upgrades[4];
+    u8 questItems[4];
+    u8 gsTokens[2];
+
+    equipment[0] = gSaveContext.inventory.items[RBA_EQUIPMENT_OFFSET + 1];
+    equipment[1] = gSaveContext.inventory.items[RBA_EQUIPMENT_OFFSET + 0];
+    gSaveContext.inventory.equipment = *(u16*)equipment;
```

Comment:
So if I understand this right, these RBA offset indices into `inventory.items` is essentially just a pointer to the other values (i.e. directly to `inventory.equipment`). Basically you are performing OOB reads to construct the byteswap value.

It would probably be better to just use the original value directly to make the byteswap instead, this way the "fix" code isn't relying on undefined behavior like OOB reads.

```c
gSaveContext.inventory.equipment = ((gSaveContext.inventory.equipment & 0xFF) >> 8) | (gSaveContext.inventory.equipment << 8)
```
And can use macros for this, one for byteswap16 and another for byteswap32

Reply:
I'm not totally a fan of using bitwise stuff here for readability sake but yea I guess I could do something like that, I guess I just didn't see much use since rba is already using two OOB reads when it happens. I think instead the safest approach would be to make a copy of Inventory as a u8 array and then byteswap and call `newInventory[gSaveContext.equips.cButtonSlots[button - 1]] = item;` and swap again after, so that way the main inventory will never be left with problematic values, but again it seems kinda messy, and if you wanted to check which part is being edited and only swap those ones it goes back to area of seeming hardcoded

Reply:
I'm not saying to use the bitwise inline (that was just an example), but rather macro it. We basically already defined macros like that for the built-in extractor here: https://github.com/HarbourMasters/Shipwright/blob/b1f0c964d29ff5edc72fe6fad894b01196fa1da0/soh/soh/Extractor/Extract.cpp#L18-L30

Readability wise, it makes more sense to perform the byteswap on the value itself rather than using OOB offsets.